src/codecs/pnm/decoder.rs

```diff
@@ -604,12 +647,17 @@ impl<R: Read> PnmDecoder<R> {
                 if bytes.len() != bytecount {
                     return Err(DecoderError::InputTooShort.into());
                 }
-
-                S::from_bytes(&bytes, width * components, buf)
-            }
-            SampleEncoding::Ascii => {
-                self.read_ascii::<S>(buf)
+                let row_size = match width.checked_mul(components.try_into().unwrap()) {
+                    Some(n) => n,
+                    None => return Err(ImageError::IoError(io::ErrorKind::UnexpectedEof.into())),
```

Comment:
This error kind doesn't really fit the failure. It's not caused by the environment, so it should be a `DecodingError` of some sort. Feel free to add one to variant to `DecoderError` in this file if none of them apply.